src/onboarding/onboarding-restore-snapshot.ts

```diff
@@ -58,11 +65,13 @@ class OnboardingRestoreSnapshot extends ProvideHassLitMixin(LitElement) {
   private async _checkRestoreStatus(): Promise<void> {
     if (this.restoring) {
       try {
-        const response = await fetch("/api/hassio/supervisor/info", {
-          method: "GET",
-        });
-        if (response.status === 401) {
-          // If we get a unauthorized response, the restore is done
+        const response = await fetchDiscoveryInformation();
+
+        if (
+          !this.discoveryInformation ||
+          this.discoveryInformation.uuid !== response.uuid
+        ) {
+          // When the UUID changes, the restore is complete
           navigate("/", { replace: true });
           location.reload();
```

Comment:
Why use `navigate` if we are going to reload anyway? Can't we use `window.location.replace`?